pyiceberg/catalog/rest.py

```diff
@@ -438,6 +444,16 @@ def _response_to_table(self, identifier_tuple: Tuple[str, ...], table_response:
             catalog=self,
         )
 
+    def _refresh_token(self) -> None:
```

Comment:
It would be nice if we can merge this logic with the logic in `_create_session`

Reply:
Good point. Changed in the new revision.